### ISSUE TITLE
Telegraf not restarting when OPC UA server restarts

### ISSUE THREAD

**Issue by john-heywood** [influxdata/telegraf#15074]
### Relevant telegraf.conf

```toml
n/a
```

### Logs from Telegraf

```text
not available
```

### System info

Telegraf 1.29.0, Telegraf 1.27.1, Telegraf 1.27.2

### Steps to reproduce

1.  Restart OPC UA server while a Telegraf v1.27.2+ is running.

### Expected behavior

When downloading to a PLC running an OPC UA server, I expected the server to restart, Telegraf recognize the change an disconnect, Telegraf reconnect, and Telegraf returning to reading data from the PLC OPC UA server. 

### Actual behavior

When the OPC UA server restarts, Telegraf loses connection to all the nodes, giving "StatusBadNodeIDUnkown" error messages. This will continue indefinitely until Telegraf is manually restarted, after which it runs as expected. 

### Additional info

This issue popped up only after upgrading to Telegraf v1.27.2 The version description says that some dependencies were updated related to OPC UA. When I downgrade to Telegraf v1.27.1 this problem does not occur. I also have an instance of Telegraf v1.29.0 where this has occurred.

**Comment by powersj** [comment]
Hi,

Without logs or a config it is hard to say for sure, but this could also be a duplicate of #13296. You can check out [this comment](https://github.com/influxdata/telegraf/issues/13296#issuecomment-1564792233) on some of the work that would be required to enable listening for nodes.

> When I downgrade to Telegraf v1.27.1 this problem does not occur.

The big change was https://github.com/influxdata/telegraf/pull/13514 which was indented to ensure we were reconnected. Again without seeing logs to understand what is going on I'm not sure what we can do.

**Comment by john-heywood** [comment]
I only have screenshots, sorry. But, here they are. We are not using a listener, just the regular OPC UA input. 
![log_issue](https://github.com/influxdata/telegraf/assets/109172534/de5e757f-7e7a-4f1c-819b-d9fa6226a78e)
^ this shows the error messages we have been seeing in versions v1.27.2+ of Telegraf not reconnecting to the OPC UA server. 
![log_v1_27_1_correct_restart](https://github.com/influxdata/telegraf/assets/109172534/c31478fc-18f2-4d36-9dd5-25a2f8fe5f6c)
^ this shows the correct reconnection that occurs with Telegraf v1.27.1
1. OPC UA server restarts
2. Telegraf disconnects it realizes something has changed
3. Telegraf reconnects
4. data is read successfully

**Comment by powersj** [comment]
@LarsStegman,

This looks like a duplicate of #13296, but what I don't understand is why the error message from OPCUA changed from bad server not connected to bad node id? Is that due to an update in the library with how it handles errors?

If we are getting a bad node id should we assume we are not connected and re-connect?

**Comment by LarsStegman** [comment]
@powersj 

> Is that due to an update in the library with how it handles errors?

I think this is the only explanation, since nothing else changed in the opc ua code. I noticed the the version we use is quite outdated already, so maybe we should consider updating soon(ish)

> If we are getting a bad node id should we assume we are not connected and re-connect?

I would prefer not to. In theory, if the node id is bad, reconnecting to the server shouldn't make a difference. In practice, opc ua is a super complex standard and reconnecting is easier than properly fixing the issue. Sometimes server are not implemented properly, are maybe we set up the connection improperly. Unfortunately, without access to the server to test against, this will be very hard to reproduce. 

@john-heywood are you able to reproduce this issue with the open62541/open62541 docker image we use for the unit tests?

**Comment by powersj** [comment]
> I noticed the the version we use is quite outdated already, so maybe we should consider updating soon(ish)

Our [gopcua in master is at 0.5.3](https://github.com/influxdata/telegraf/blob/master/go.mod#L97), which looks to be the [latest](https://github.com/gopcua/opcua/releases)? That went out in v1.30.

**Comment by jyarr** [comment]
I have replicated the issue on an OPC UA server hosted on the same manufacture hardware.

Siemens Model: S7-1515F-2PN, Article No:6ES7515-2FM02-0AB0, Firmware 2.9.7.

Telegraf reconnects in v1.27.1 and does not successfully reconnect in v1.27.2+ including the latest v1.30.0 that updated to gopcua 0.5.3.

Attached is my telegraf.conf, telegraf logs, and wire shark captures of successful reconnects in 1.27.1 and failure to reconnect in 1.27.2.

Restarting telegraf in v1.27.2+ was the only way I could get telegraf to reconnect.

[telegraf.conf.txt](https://github.com/influxdata/telegraf/files/14782149/telegraf.conf.txt)

[telegrafLog_v1_27_2_UnsuccessfulReconnect_TelegrafRestartRequired.txt](https://github.com/influxdata/telegraf/files/14782167/telegrafLog_v1_27_2_UnsuccessfulReconnect_TelegrafRestartRequired.txt)

[telegrafLog_v1_27_1_SuccessfulReconnect.txt](https://github.com/influxdata/telegraf/files/14782152/telegrafLog_v1_27_1_SuccessfulReconnect.txt)


[telegraf_v1_27_2_opcua_wireshark_capture_UnsuccessfulReconnect_FixedAfterTelegrafRestart.pcapng.gz](https://github.com/influxdata/telegraf/files/14782179/telegraf_v1_27_2_opcua_wireshark_capture_UnsuccessfulReconnect_FixedAfterTelegrafRestart.pcapng.gz)

[telegraf_v1_27_1_opcua_wireshark_capture_SuccessfulReconnect.pcapng.gz](https://github.com/influxdata/telegraf/files/14782180/telegraf_v1_27_1_opcua_wireshark_capture_SuccessfulReconnect.pcapng.gz)
